src/ucp/core/ucp_ep.c

```diff
@@ -72,7 +73,24 @@ ucs_status_t ucp_ep_new(ucp_worker_h worker, uint64_t dest_uuid,
     ep->cfg_index        = ucp_worker_get_ep_config(worker, &key);
     ep->am_lane          = UCP_NULL_LANE;
     ep->flags            = 0;
-    ucs_queue_head_init(&ep->stream_data);
+
+    if (worker->context->config.features & UCP_FEATURE_STREAM) {
```

Comment:
maybe you could define ucp_stream_ep which inherits ucp_ep (and allocate it if UCP_FEATURE_STREAM)? You would not need ep->ext.stream->ucp_ep then

Reply:
in addition it will not increase ucp ep size if 'stream' feature is not requested and removing extra pointer reference will be more cache friendly.

Reply:
The idea is to make it extendable without base EP growing. For example, user can use STREAM and TAG on the same EP, then we will allocate only corresponding extensions independently. Current EP implementation is too "AM and TAG oriented".

Reply:
ok